easybuild/easyconfigs/i/ICON/ICON-2024.01-foss-2023a.eb

```
@@ -0,0 +1,217 @@
+# This file is an EasyBuild recipe as per https://easybuild.io
+# Author: Jens Henrik Göbbert
+# Forschungszentrum Juelich GmbH (FZJ)
+# Juelich Supercomputing Centre (JSC)
+
+easyblock = 'ConfigureMake'
+
+name = 'ICON'
+version = '2024.01'
+
+local_tag = 'icon-%(version)s-public'
+
+homepage = "https://www.icon-model.org/"
+description = """
+ICON is a flexible, scalable, high-performance modelling framework for weather, climate and environmental prediction
+that provides actionable information for society and advances our understanding of the Earth's climate system.
+"""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'pic': True, 'usempi': True}
+
+source_urls = [('https://gitlab.dkrz.de/icon/icon-model/-/archive/release-%(version)s-public/')]
+sources = [('icon-model-release-%(version)s-public.tar.gz')]
+
+patches = [
+    'ICON-2024.01_lzacc-undefined.patch',
+]
+checksums = [
+    '53f0c6fb2622725b184b220ace307665d7965ae3609ebe4a51e5e36539a495b0',
+    '5612f304dc0241016e3126134737549d7bb719cb879451ef3a9623a29a218414',
+]
+
+builddependencies = [
+    ('CMake', '3.26.3'),
+    ('Python', '3.11.3'),
+    ('SciPy-bundle', '2023.07', '', ('gfbf', '2023a')),
+    ('Perl', '5.36.1'),
+    ('pkgconf', '1.9.5'),
+]
+
+dependencies = [
+    ('zlib', '1.2.13'),
+    ('libxml2', '2.11.4'),
+    ('libfyaml', '0.9'),
+    ('ecCodes', '2.31.0'),
+    ('netCDF', '4.9.2'),
+    ('netCDF-Fortran', '4.6.1'),
+    ('HDF5', '1.14.3'),
```

In `foss/2023a`, this HDF5 version is used:
```suggestion
    ('HDF5', '1.14.0'),
```

---

Thanks. I fixed it.